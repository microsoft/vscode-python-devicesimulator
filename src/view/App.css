.App {
  text-align: center;
  display: flex;
  justify-content: space-between;
}

.App-main {
  background-color: var(--vscode-editor-background);
  padding: 0px 0.75em 0px 0.75em;
  min-height: 100vh;
  width: 100%;
  margin-top: 51px;
  margin-bottom: 53px;
  max-height: 400px;
  overflow: scroll;
}
