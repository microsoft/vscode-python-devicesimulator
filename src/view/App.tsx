// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import * as React from "react";
import "./App.css";
import {
    DEVICE_LIST_KEY,
    VIEW_STATE,
    VSCODE_MESSAGES_TO_WEBVIEW,
    WEBVIEW_ATTRIBUTES_KEY,
    WEBVIEW_TYPES,
} from "./constants";
import { Device } from "./container/device/Device";
import { ViewStateContext } from "./context";
import { GettingStartedPage } from "./pages/gettingStarted";

interface IState {
    currentDevice: string;
    viewState: VIEW_STATE;
    type?: WEBVIEW_TYPES;
}

const defaultState = {
    currentDevice: DEVICE_LIST_KEY.CPX,
    viewState: VIEW_STATE.RUNNING,
    type: undefined,
};

class App extends React.Component<{}, IState> {
    constructor() {
        super({});
        this.state = defaultState;
    }
    componentDidMount() {
        if (document.currentScript) {
            const webviewTypeAttribute = document.currentScript.getAttribute(
                WEBVIEW_ATTRIBUTES_KEY.TYPE
            ) as WEBVIEW_TYPES;
            if (webviewTypeAttribute) {
                this.setState({ type: webviewTypeAttribute });
            }
            if (webviewTypeAttribute === WEBVIEW_TYPES.SIMULATOR) {
                const initialDevice = document.currentScript.getAttribute(
                    WEBVIEW_ATTRIBUTES_KEY.INITIAL_DEVICE
                );

                if (initialDevice) {
                    this.setState({ currentDevice: initialDevice });
                    window.addEventListener("message", this.handleMessage);
                }
            }
        }
    }
    componentWillUnmount() {
        window.removeEventListener("message", this.handleMessage);
    }

    render() {
        return (
            <div className="App">
                <main className="App-main">
                    <ViewStateContext.Provider value={this.state.viewState}>
                        {this.loadContent()}
                    </ViewStateContext.Provider>
                </main>
            </div>
        );
    }
    loadContent = () => {
        console.log(this.state.type);
        switch (this.state.type) {
            case WEBVIEW_TYPES.GETTING_STARTED:
                return <GettingStartedPage />;
            case WEBVIEW_TYPES.SIMULATOR:
                return (
                    <Device currentSelectedDevice={this.state.currentDevice} />
                );
        }
        return;
    };

    handleMessage = (event: any): void => {
        const message = event.data;

        switch (message.command) {
            case VSCODE_MESSAGES_TO_WEBVIEW.SET_DEVICE:
                if (message.active_device !== this.state.currentDevice) {
                    this.setState({ currentDevice: message.active_device });
                }
                break;
            case VSCODE_MESSAGES_TO_WEBVIEW.RUN_DEVICE:
                this.setState({ viewState: VIEW_STATE.RUNNING });
                break;
            case VSCODE_MESSAGES_TO_WEBVIEW.PAUSE_DEVICE:
                this.setState({ viewState: VIEW_STATE.PAUSE });
                break;
        }
    };
}

export default App;
