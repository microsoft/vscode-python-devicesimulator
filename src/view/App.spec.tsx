import * as React from "react";
import * as ReactDOM from "react-dom";
import { IntlProvider } from "react-intl";
import * as testRenderer from "react-test-renderer";
import App from "./App";

describe("App component ", () => {
    it("should render correctly", () => {
        const component = testRenderer
            .create(
                <IntlProvider locale="en">
                    <App />
                </IntlProvider>
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <IntlProvider locale="en">
                <App />
            </IntlProvider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });
});
