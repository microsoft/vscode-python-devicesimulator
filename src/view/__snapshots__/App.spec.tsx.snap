// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component  should render correctly 1`] = `
<div
  className="App"
>
  <main
    className="App-main"
  />
</div>
`;
